content/blog/overview-package-management-python.md

```diff
@@ -0,0 +1,363 @@
+```
+
+As opposed to the other tools, `poetry` uses the semantic versioning to refer to dependencies versions. If you have used
+[Cargo](https://github.com/rust-lang/cargo) in Rust, you will feel at home. The issue is that many Python libraries do not use
+semantic versioning so it does feel a bit weird to use in this context.
+
+While trying to clean its cache to test the speed, I ran into a small UX issue.
+Poetry offers a command to clean the cache (`poetry cache:clear`) but it takes a path/string to a cache as an argument. A new
+user like me will not know what argument to give, especially since the command is not [in the documentation](https://poetry.eustace.io/docs/cli/).
+Looking into issues, we can see that we can run `poetry cache:clear --all pypi` but the documentation doesn't even mention `cache:clear` so it's not
+obvious.
+
+In terms of speed, a `poetry install` with the same list of packages as the others was done in 40s.
+Subsequent runs are below a second. The UI is very clean and clear about what was installed and their versions.
+
+Poetry also has `poetry update` to update dependencies and once again makes it clear what has changed.
+
+While this post is focused on backend applications, it is worth mentioning that Poetry apparently handles building and publishing
+libraries as well which is a big plus if you are a maintainer: one tool for all purposes!
+
+## My opinion
+
+I have only mentioned 3 tools but there are way more! I wouldn't be surprised if Python had the record of the highest
+number of tools dedicated to package and environment management for a programming language.
+Since all those tools are written in Python themselves you still need to use Pip first to install them, globally sometimes. A better solution
+in my opinion would be to write the package manager in a language compiling down to a binary, solving the bootstrapping issue and
+being a good example of https://xkcd.com/927/.
+
+To go back to the three tools mentioned in this article, I would recommend either pip-tools or poetry.
+My limited usage of Pipenv had too many WTFs to be considered ready for actual use, not even taking into account
+its extreme slowness. I'm not entirely sure why it is the recommended tool other than the virtualenv setup was easy to do because it feels
+like an alpha version. If you are Docker containers for your development, then I don't see any advantage of Pipenv other pip-tools or Poetry.
+
+I like the simplicity of `pip-tools`. As it is only a small step to take if you are writing `requirements.txt` by hand currently, I
+would probably recommend that right now.
+
+On the other hand, if `pyproject.toml` gets adopted by other tools (Flake8, mypy etc), having everything in one file would be very
+appealing and poetry would become the best choice at that time. I expect `poetry` to work with my current use of `virtualenvwrapper` though,
+I don't want to have to convert all my projects to poetry just to make it happy.
```

Comment:
Sometimes you can't. while working on someone else's project, you can't change the package management